sklearn/datasets/tests/test_openml.py

```diff
@@ -1216,15 +1216,16 @@ def test_fetch_openml_raises_missing_values_target(monkeypatch, gzip_response):
 
 
 def test_fetch_openml_raises_illegal_argument():
-    msg = 'Dataset data_id='
+    msg = 'Dataset data_id=-1 and version=version passed, but you can only'
     with pytest.raises(ValueError, match=msg):
-        fetch_openml(data_id=-1, name="name")
+        fetch_openml(data_id=-1, name=None, version="version")
 
+    msg = "Dataset data_id=-1 and name=name passed, but you can only"
     with pytest.raises(ValueError, match=msg):
-        fetch_openml(data_id=-1, name=None, version="version")
+        fetch_openml(data_id=-1, name="nAmE")
 
     with pytest.raises(ValueError, match=msg):
-        fetch_openml(data_id=-1, name="name", version="version")
+        fetch_openml(data_id=-1, name="nAmE", version="version")
```

Comment:
I do not think we should change the original tests. We can add an additional test case that is similar to the first one you have:

```python
msg = 'Dataset data_id=-1 and version=version passed, but you can only'
with pytest.raises(ValueError, match=msg):
    fetch_openml(data_id=-1, version="version")
```

Reply:
This change only reorders tests and makes them more strict by including the actual `name` / `data_id` / `version` in the error message to match.